src/main/java/org/openrewrite/micrometer/TimerToObservation.java

```diff
@@ -0,0 +1,144 @@
+                                    builder.add("\n.lowCardinalityKeyValue(#{any(String)}, #{any(String)})");
+                                    parameters.add(builderMethod.getArguments().get(0));
+                                    parameters.add(builderMethod.getArguments().get(1));
+                                }
+                                else if (tagsIterableMatcher.matches(maybeBuilder)) {
+                                    builder.add("\n.lowCardinalityKeyValues(KeyValues.of(#{any(Iterable)}, Tag::getKey, Tag::getValue))");
+                                    parameters.addAll(builderMethod.getArguments());
+                                }
+                                else if (tagsMatcher.matches(maybeBuilder)) {
+                                    String args = StringUtils.repeat("#{any(String)},", builderMethod.getArguments().size());
+                                    args = args.substring(0, args.length() - 1);
+                                    builder.add("\n.lowCardinalityKeyValues(KeyValues.of(" + args + "))");
+                                    parameters.addAll(builderMethod.getArguments());
+                                }
+                                maybeBuilder = ((J.MethodInvocation) maybeBuilder).getSelect();
+                            }
+                            if (timerName != null) {
+                                parameters.add(0, timerName);
+                                parameters.add(1, registry);
+
+                                maybeAddImport("io.micrometer.observation.Observation");
+                                maybeAddImport("io.micrometer.common.KeyValues");
+                                maybeAddImport("io.micrometer.core.instrument.Tag");
+                                maybeRemoveImport("io.micrometer.core.instrument.Timer");
+
+                                JavaTemplate template = JavaTemplate
+                                        .builder("Observation.createNotStarted(#{any(java.lang.String)}, #{any()})"
+                                                + String.join("", builder))
+                                        .contextSensitive()
+                                        .javaParser(JavaParser.fromJavaVersion()
+                                                .classpathFromResources(executionContext, "micrometer-observation", "micrometer-commons", "micrometer-core"))
+                                        .imports("io.micrometer.observation.Observation")
+                                        .imports("io.micrometer.common.KeyValues")
+                                        .imports("io.micrometer.core.instrument.Tag")
+                                        .build();
+
+                                mi = autoFormat(
+                                        template.apply(updateCursor(mi), mi.getCoordinates().replace(), parameters.toArray()),
+                                        executionContext
```

Comment:
Would maybe be good to at least try to retain any comments attached to the Timer, as that's where I most expect them to be.

Reply:
The comments are maintained since they are attached to the last method invocation record -> observe, and we change this one with the ChangeMethodName recipe, so it is maintained. Do you think it makes sense to try to keep also the comments between method invocations? The only one being removed right now are the "Comments on register", and I'm having a hard time to decide where they should be in the result:
```
// Comments on Timer
Timer.builder("my.timer")
       // Comments on register
       .register(registry)
       // Comments on record
       .record(arg);
```
```
// Comments on Timer
Observation.createNotStarted("my.timer", registry)
        // Comments on record
        .observe(arg);
```

Reply:
No need to keep the comments between other method invocations; this should work well enough for now.